components/builder-originsrv/src/server/handlers.rs

```diff
@@ -608,6 +608,23 @@ pub fn origin_package_list(
     Ok(())
 }
 
+pub fn origin_packages_for_version(
```

Comment:
I generally like `with_channels` over `for_version`, yeah -- I'll make this change everywhere.